src/nwm_routing/src/nwm_routing/preprocess.py

```diff
@@ -1,11 +1,26 @@
 import time
 import pandas as pd
+import xarray as xr
+from datetime import datetime
 from collections import defaultdict
 import troute.nhd_network_utilities_v02 as nnu
 import troute.nhd_network as nhd_network
 import troute.nhd_io as nhd_io
 
 
+def get_t0_str_from_restart_parameters(
+    restart_parameters,
+):
+    if restart_parameters.get("wrf_hydro_channel_restart_file",None):
+        channel_initial_states_file = restart_parameters["wrf_hydro_channel_restart_file"]
+        with xr.open_dataset(channel_initial_states_file) as ds:
+            t0_str = ds.Restart_Time.replace("_", " ")
+    else:
+        t0_str = "2015-08-16 00:00:00"
```

Comment:
If we don't have a restart file, I suppose we are just cold starting from an arbitrary t0 (this date). Do we have the capability to read restarts from csv? If so, do those files contain any sort of start information?

Reply:
The restart from CSV capability permits but does not require date-centric headers for the CSV columns. The two places that the t0 is used directly are 1: for the Reservoir simulation; 2: for dealing with the restart file write out. 

We could add a parameter to the yaml file to allow the user to directly specify a t0. That might be used to provide a value to accompany a csv file lacking the information otherwise. It might also, either intentionally or accidentally, override the value in the file. 

My recommendation is to avoid over complicating the input handling -- We are designing most specifically for the operational case, where a t0 will always be available and known. If we require the use to input that at the front, then it makes it easy for a later effort to create a wrapper that provides it by more sophisticated heuristics.

Reply:
Setting this out as issue #427